scripts/release/release-notes-onprem.sh

```diff
@@ -0,0 +1,207 @@
+  echo "Done."
+
+  echo "Creating ${RELEASE_VERSION} in Jira..."
+  payload_version="{\"archived\":false,\"name\":\"${RELEASE_VERSION}\",\"projectId\":${JIRA_PROJECT_ID},\"released\":false,\"description\":\"OnPrem Release for ${RELEASE_VERSION} by $JIRA_USER\"}"
+  payload_issue="{\"update\": {\"fixVersions\": [{\"add\": {\"name\": \"${RELEASE_VERSION}\"}}]}}"
+  curl -s -o /dev/null --request POST \
+     --url "$JIRA_URL/rest/api/3/version" \
+     --user "$JIRA_AUTH" \
+     --header 'Accept: application/json' \
+     --header 'Content-Type: application/json' \
+     --data "${payload_version}"
+  echo "Done."
+fi
+
+
+####################
+# Find all related tickets
+####################
+fixes=""
+features=""
+migrations=""
+review=""
+
+echo "Retreiving related tickets..."
+for i in $(git log $PROD_TAG...$DEV_TAG | grep -Eo '(PACT-)([0-9]+)' | sort | uniq); do
+   response=$(curl -s --request GET \
+       --url "$JIRA_URL/rest/api/3/issue/$i?fields=customfield_11009,customfield_18528,customfield_17522,status" \
+       --user "$JIRA_AUTH" \
+       --header 'Accept: application/json' \
+       --header 'Content-Type: application/json')
+   
+   # Valid Scenario: We may have merged code for a ticket that has not been marked as done in Jira since there is 
+   # more work required to complete it but we still want to attach a version to the ticket which tells Jira that 
+   # part of the code has been released to production.
+   status=$(echo $response | jq '.fields.status.name' | tr -d '"')
+   if [ "$status" = "Done" ]; then
+
+       platform=$(echo $response | jq '.fields.customfield_17522.value' | tr -d '"')
+       if [ "$platform" = "saas" ]; then
+          continue
+       fi
+
+       release_type=$(echo $response | jq '.fields.customfield_18528.value' | tr -d '"')
+       has_note=$(echo $response | jq '.fields.customfield_11009' | tr -d '"')
+
+       if [ "$release_type" = "Feature" ] && [ "$has_note" != "null" ]; then
+         features+="\n- "$(echo $response | jq '.fields.customfield_11009.content[].content[].text' | tr -d '"')
+       elif [ "$release_type" = "Fix" ] && [ "$has_note" != "null" ]; then
+         fixes+="\n- "$(echo $response | jq '.fields.customfield_11009.content[].content[].text' | tr -d '"')
+       else
+         review+='\n- https://smartbear.atlassian.net/browse/'$i
+       fi
+   fi
+
+   if [ -n "$IS_RELEASE" ]; then
+     curl --request PUT \
+         --url "$JIRA_URL/rest/api/3/issue/$i" \
+         --user "$JIRA_AUTH" \
+         --header 'Accept: application/json' \
+         --header 'Content-Type: application/json' \
+         --data "${payload_issue}"
+    fi
+    
+done
+echo "Done."
+
+####################
+# Write the release notes, output to screen
+####################
+release_note_file=${DOCS_ROOT_DIT}/website/docs/docs/on-premises/releases/${RELEASE_VERSION}.md
+
+echo
+echo -e "
+---
+title: ${RELEASE_VERSION}
+---
+
+## Release date
+
+$(date +"%Y-%m-%d")
```

Comment:
Our previous notes have date format `DD FULL_MONTH YYYY`. I think we should keep this format for consistency.